pkg/api/feed_test.go

```diff
@@ -0,0 +1,383 @@
+	)
+
+	t.Run("update once", func(t *testing.T) {
+		var (
+			timestamp        = int64(12121212)
+			timestampBytes   = make([]byte, 8)
+			ref              = swarm.MustParseHexAddress("accdaccd")
+			ch               = mockUpdate(timestamp, expReference)
+			_                = common.HexToAddress(ownerString)
+			look             = newMockLookup(-1, 2, ch, nil)
+			factory          = newMockFactory(look)
+			sch, owner, data = socChunk(t, ch)
+			client, _, _     = newTestServer(t, testServerOptions{
+				Storer: mockStorer,
+				Tags:   tag,
+				Feeds:  factory,
+			})
+		)
+
+		binary.BigEndian.PutUint64(timestampBytes, uint64(timestamp))
+
+		respHeaders := jsonhttptest.Request(t, client, http.MethodPut, fmt.Sprintf("/feeds/%s/%s/%s", ownerString, "aabbcc", ref.String()), http.StatusOK,
+			jsonhttptest.WithExpectedJSONResponse(api.FeedReferenceResponse{Reference: expReference}),
+		)
+
+		// find out if the soc made it into the localstore
+		cch, err := mockStorer.Get(context.Background(), storage.ModeGetRequest, sch.Address())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !bytes.Equal(cch.Data(), ch.Data()) {
+			t.Fatal("data mismatch")
+		}
+		//if h := respHeaders[api.SwarmFeedIndexHeader]; len(h) > 0 {
+		//b, err := hex.DecodeString(h[0])
+		//if err != nil {
+		//t.Fatal(err)
+		//}
+		//if !bytes.Equal(b, timestampBytes) {
+		//t.Fatalf("feed index header mismatch. got %v want %v", b, timestampBytes)
+		//}
+		//} else {
+		//t.Fatal("expected swarm feed index header to be set")
+		//}
+	})
+
+	//t.Run("malformed owner", func(t *testing.T) {
+	//jsonhttptest.Request(t, client, http.MethodGet, feedResource("xyz", "cc", "", ""), http.StatusBadRequest,
+	//jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+	//Message: "bad owner",
+	//Code:    http.StatusBadRequest,
+	//}),
+	//)
+	//})
+
+}
+
+type factoryMock struct {
+	sequenceCalled bool
+	epochCalled    bool
+	feed           *feeds.Feed
+	lookup         feeds.Lookup
+}
+
+func newMockFactory(mockLookup feeds.Lookup) *factoryMock {
+	return &factoryMock{lookup: mockLookup}
+}
+
+func (f *factoryMock) NewLookup(t feeds.Type, feed *feeds.Feed) feeds.Lookup {
+	switch t {
+	case feeds.Sequence:
+		f.sequenceCalled = true
+	case feeds.Epoch:
+		f.epochCalled = true
+	}
+	f.feed = feed
+	return f.lookup
+}
+
+type mockLookup struct {
+	at, after int64
+	chunk     swarm.Chunk
+	err       error
+}
+
+func newMockLookup(at, after int64, ch swarm.Chunk, err error) *mockLookup {
+	return &mockLookup{at: at, after: after, chunk: ch, err: err}
+}
+
+func (l *mockLookup) At(_ context.Context, at, after int64) (swarm.Chunk, error) {
+	if l.at == -1 {
+		// shortcut to ignore the value in the call since time.Now() is a moving target
+		return l.chunk, nil
+	}
+	fmt.Println(at, after, l.at, l.after)
+	if at == l.at && after == l.after {
+		return l.chunk, nil
+	}
+	return nil, errors.New("no feed update found")
+}
+
+func mockUpdate(t int64, a swarm.Address) swarm.Chunk {
```

Comment:
no need to have a mock when the mock does what the real thing does. just use the feed toChunk method etc